libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -68,8 +68,10 @@ void StaticAnalyzer::endVisit(FunctionDefinition const&)
 	m_nonPayablePublic = false;
 	m_constructor = false;
 	for (auto const& var: m_localVarUseCount)
-		if (var.second == 0)
-			m_errorReporter.warning(var.first->location(), "Unused local variable");
+		if (var.second == 0 && var.first->isCallableParameter())
+			m_errorReporter.warning(
+				var.first->location(), 
+				"Unused local variable. Remove or comment out the variable name to silence this warning");
```

Comment:
Replace `local variable` with `parameter`?

Reply:
I think there must be two messages, depending whether the variable `isCallableParameter` or not.

If it is, suggest dropping or commenting out the variable name (but keeping the variable).

If it isn't, suggest removing the variable.